server/src/main/java/io/crate/operation/collect/files/CSVLineParser.java

```diff
@@ -84,6 +104,43 @@ public void parseHeader(String header) throws IOException {
         }
     }
 
+    @Nullable
+    public byte[] parse() throws IOException {
+
+        if (parser.getInputSource() == null) {
```

Comment:
had to add this, despite on 
csvReader = csvReader.without(JsonParser.Feature.AUTO_CLOSE_SOURCE);

parser closes stream when file is read up - that's why commented out it above. 
it's actually fine behavior, we want to close when file is exhausted but would be better to control it explicitly. 
closeCurrentStream probably not needed now (falls to == null anyway). 
To sum up - it's fine only for testing perf as it doesn't break correctness, file is read up, resource are released but we need to control it explicitly in the real implementation i.e make `csvReader.without(JsonParser.Feature.AUTO_CLOSE_SOURCE);` work